kymatio/scattering2d/filter_bank.py

```diff
@@ -46,14 +46,24 @@ def filter_bank(M, N, J, L=8):
             psi_signal_fourier = fft2(psi_signal)
             # drop the imaginary part, it is zero anyway
             psi_signal_fourier = np.real(psi_signal_fourier)
-            for res in range(min(j + 1, J - 1)):
+            if J > 1:
+                for res in range(min(j + 1, J - 1)):
```

Comment:
OK so this block looks like a copy/paste of block below.

Is it not possible to find a way of parametrizing the loop such that the J=1 case is contained within it the way we want?

If not, then I'm OK with duplicating code (I'd like to avoid it if it is easily avoided though)